src/nbl/video/CVulkanCommandBuffer.h

```diff
@@ -1398,6 +1398,8 @@ class CVulkanCommandBuffer : public IGPUCommandBuffer
 
         CVulkanCommandPool* vulkanCommandPool = IBackendObject::compatibility_cast<CVulkanCommandPool*>(m_cmdpool.get(), this);
         vulkanCommandPool->emplace_n(m_argListTail, begin, end);
+        // TODO: verify this
+        if (!m_argListHead) m_argListHead = m_argListTail;
```

Comment:
What if this for? 
Did your write this @deprilula28 ?

Reply:
that's part of the command buffer references fix, without this freeSpaceInCmdPool will never do anything as m_argListHead is always null